### PR TITLE
add instance types to datastore metrics

### PR DESCRIPTION
**Pull request by bhowmickatish**
### What does this PR do?

https://jira.nutanix.com/browse/ENG-379748

A brief description of the change being made with this pull request.

### Motivation
Add file types as instance value to query datastore metrics 

What inspired you to submit this pull request?
https://jira.nutanix.com/browse/ENG-379748

### Additional Notes

Anything else we should know when reviewing?
AOC repo changes will b done in a separate PR . 

3 new datatstore metrics will be generated : 
disk.used.latest.diskfile
disk.used.latest.deltafile
disk.used.latest.otherfile

disk.used.latest -> existing metrics will continue to have the aggregated counter

### CODE REVIEW

File: vsphere/datadog_checks/vsphere/metrics.py [hunk 1]
```diff
@@ -82,7 +82,7 @@
     'datastore.numberWriteAveraged.average' : None,
     'disk.capacity.latest' : None,
     'disk.provisioned.latest' : None,
-    'disk.used.latest' : None,
+    'disk.used.latest' : ",DISKFILE,DELTAFILE,OTHERFILE",
```
Comment (ahalam) [inline vsphere/datadog_checks/vsphere/metrics.py:85]: is the `,` at the start of string intentional?

Comment (bhowmickatish) [reply vsphere/datadog_checks/vsphere/metrics.py:85]: yeah it is ... it signifies an empty instance value which queries the aggregated metric counter

File: vsphere/datadog_checks/vsphere/vsphere.py [hunk 2]
```diff
@@ -1302,7 +1304,13 @@ def _collect_metrics_atomic(self, instance, query_specs):
                                 self.log.debug(u"Skipping `%s` metric because the value is empty", metric_name)
                                 continue
 
-                            instance_name = perf_metric.id.instance or "none"
+                            metric_instance = str(perf_metric.id.instance)
+                            instance_name = metric_instance or "none"
+                            # append metric name with valid instance value for datastore metrics
```
Comment (ahalam) [inline vsphere/datadog_checks/vsphere/vsphere.py:1309]: ~We used to report overall metrics but they will be report per `DATASTORE_METRIC_FILE_TYPES`. If any one wants overall metrics, will they will have to aggregate them?~

Comment (bhowmickatish) [reply vsphere/datadog_checks/vsphere/vsphere.py:1309]: we have retained the aggregate query as well ... just added 3 file types. so basically now 4 metrics will b reported : 
existing -> disk.used.latest (aggregate) 
new metrics : 
disk.used.latest.diskfile
disk.used.latest.deltafile
disk.used.latest.otherfile

Comment (bhowmickatish) [reply vsphere/datadog_checks/vsphere/vsphere.py:1309]: hence no impact on existing functionality ... UI can now query these 3 additional metrics & display by file type

**Review by ahalam — APPROVED** [review]

**Pull request merged**